server/server.cc

```diff
@@ -0,0 +1,77 @@
+
+// based off of: http://www.boost.org/doc/libs/1_54_0/doc/html/boost_asio/example/cpp11/echo/async_tcp_echo_server.cpp
+// http://think-async.com/asio/boost_asio_1_5_3/doc/html/boost_asio/example/local/stream_server.cpp
+
+#include <cstdlib>
+#include <iostream>
+#include <memory>
+#include <utility>
+#include <boost/asio.hpp>
+
+#include <vector>
+#include "server.h"
+
+using boost::asio::ip::tcp;
+
+Session::Session(tcp::socket socket)
+  : socket_(std::move(socket)) {
+}
+
+void Session::start() {
+  do_read();
+}
+
+void Session::do_read() {
+  auto self(shared_from_this());
+  data_ = std::string(2048, 0);
```

Comment:
Will there be a problem if the size of the input is bigger than 2048? ThoughI don't know how they will test it...